src/tools/tool-options.h

```diff
@@ -24,98 +24,109 @@
+                       "Enable all features "
+                       "(default if there is no target features section"
+                       " or if any feature options are provided)",
+                       [](FeatureSet& features) {
+                         features = FeatureSet::All;
+                       })
+        .addFeatureOpt("--detect-features", "",
+                       "Use features from the target features section"
+                       "(default if there is a target features section"
+                       " and no feature options are provided)",
+                       [this](FeatureSet& features) {
+                         features = wasmFeatures;
+                         errorOnFeatureMismatch = false;
+                       })
+        .addFeature("sign-ext", "sign extension operations",
+                    FeatureSet::SignExt)
+        .addFeature("threads", "atomic operations",
+                    FeatureSet::Atomics)
+        .addFeature("mutable-globals", "mutable globals",
+                    FeatureSet::MutableGlobals)
+        .addFeature("nontrapping-float-to-int",
+                    "nontrapping float-to-int operations",
+                    FeatureSet::TruncSat)
+        .addFeature("simd", "SIMD operations and types",
+                    FeatureSet::SIMD)
+        .addFeature("bulk-memory", "bulk memory operations",
+                    FeatureSet::BulkMemory)
+        .add("--no-validation", "-n",
+             "Disables validation, assumes inputs are correct",
              Options::Arguments::Zero,
              [this](Options* o, const std::string& argument) {
                passOptions.validate = false;
              });
-        ;
   }
 
-  FeatureSet getFeatures() const {
+  ToolOptions &addFeatureOpt(const std::string& longName,
+                             const std::string& shortName,
+                             const std::string& description,
+                             Builder builder) {
+    add(longName, shortName, description, Options::Arguments::Zero,
+        [=](Options*, const std::string&) {
+          featureBuilders.push_back(builder);
+        });
+    return *this;
+  }
+
+  ToolOptions &addFeature(const std::string& name,
+                          const std::string& description,
+                          FeatureSet::Feature feature) {
+    (*this)
+        .addFeatureOpt(std::string("--enable-") + name, "",
+                       std::string("Enable ") + description,
+                       [=](FeatureSet& features) {
+                         features.set(feature, true);
+                       })
+        .addFeatureOpt(std::string("--disable-") + name, "",
+                       std::string("Disable ") + description,
+                       [=](FeatureSet& features) {
+                         features.set(feature, false);
+                       });
+    return *this;
+  }
+
+  FeatureSet& getFeatures(const Module& module) {
```

Comment:
As mentioned earlier I'd feel better about a `computeFeatures(Module&)` that does the computation and a `getFeatures()` that just immediately returns them.